libbeat/autodiscover/autodiscover.go

```diff
@@ -143,37 +139,39 @@ func (a *Autodiscover) worker() {
 
 			if _, ok := event["start"]; ok {
 				updated = a.handleStart(event)
+				kind = "start"
 			}
 			if _, ok := event["stop"]; ok {
 				updated = a.handleStop(event)
+				kind = "stop"
 			}
 
-		case <-time.After(retryPeriod):
-		}
-
-		if updated || retry {
-			if retry {
-				a.logger.Debug("Reloading existing autodiscover configs after error")
-			}
+		case <-t.C:
+			if updated || retry {
+				a.logger.Debugf("Reloading autodiscover configs reason: updated: %t, retry: %t, kind: %s", updated, retry, kind)
 
-			configs := []*reload.ConfigWithMeta{}
-			for _, list := range a.configs {
-				for _, c := range list {
-					configs = append(configs, c)
+				configs := []*reload.ConfigWithMeta{}
+				for _, list := range a.configs {
+					for _, c := range list {
+						configs = append(configs, c)
+					}
 				}
-			}
 
-			err := a.runners.Reload(configs)
+				a.logger.Debugf("calling reload with %d config(s)", len(configs))
+				err := a.runners.Reload(configs)
 
-			// On error, make sure the next run also updates because some runners were not properly loaded
-			retry = err != nil
-			// reset updated status
-			updated = false
-		}
+				// reset updated status
+				updated = false
+
+				// On error, make sure the next run also updates because some runners were not properly loaded
+				retry = err != nil
+				if retry {
+					t.Reset(10 * debouncePeriod)
```

Comment:
Might be good to add a comment here explaining why we reset the timer to 10x the original period in case of an error.